### PR TITLE
Conan deps build

### PR DESCRIPTION
**Pull request by pawsten**
* All dependency included in conan. Conan build   with  cmake parameter -DCONAN_DEPS=1
* Cleaned up all mess with link libraries
* Fixed  possibility of deadlock  during sending SIGTERM to nettracer in case of writing larger portion of data to stdout
* Fixed correct headers to building bpf program

### CODE REVIEW

**Comment by pijuszczyk** [comment]
I've added a few comments here and there but all in all thanks for adding conan! That's awesome

File: nettracersrv/nettracer.cpp [hunk 1]
```diff
@@ -32,6 +32,8 @@ static ExitCtrl exitCtrl;
 
 void atexit_handler(int a) {
 	exitCtrl.running = false;
+	close(0);
+	close(1);
```
[the remark this answers is not in the document]
Comment (pawsten) [reply nettracersrv/nettracer.cpp:36]: "0" is fd of stdin, "1" of stdout. https://man7.org/linux/man-pages/man3/stdout.3.html

File: CMakeLists.txt [hunk 2]
```diff
@@ -67,17 +74,29 @@ endif()
...
 # - the one related to the current version & arch
 # - and possibly another one for generic headers
-function(try_adding_kernel_headers_base_dir out_dirs suffix)
+function(try_adding_kernel_headers_base_dir out_dirs suffix mod_include)
 	string(REGEX REPLACE "-[^-]+$" "-${suffix}" MOD_KERNEL_VERSION "${KERNEL_VERSION}")
 	if(NOT MOD_KERNEL_VERSION STREQUAL KERNEL_VERSION)
 		set(DIR "/usr/src/linux-headers-${MOD_KERNEL_VERSION}")
 		if(EXISTS "${DIR}")
-			set(${out_dirs} "${${out_dirs}};${DIR}" PARENT_SCOPE)
+			set(${mod_include} "${DIR}/include/linux" PARENT_SCOPE)
```
[the remark this answers is not in the document]
Comment (pawsten) [reply CMakeLists.txt:85]: I've removed reduntant parameter from try_adding_kernel_headers_base_dir

File: CMakeLists.txt [hunk 3]
```diff
@@ -102,7 +121,14 @@ arch/${ARCH}/include/generated/uapi")
 endfunction()
 collect_kernel_headers_dirs("${KERNEL_HEADERS_BASE}" KERNEL_HEADERS)
 
-set(GCC_HEADERS "/usr/lib/gcc/${ARCHITECTURE}-linux-gnu/${GCC_VERSION}/include")
```
[the remark this answers is not in the document]
Comment (pawsten) [reply CMakeLists.txt:105]: I've removed all instances of GCC_VERSION

File: nettracersrv/nettracer.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (pijuszczyk) [reply nettracersrv/nettracer.cpp:36]: Thanks

**Comment by pijuszczyk** [comment]
All seems good :)

**Review by pijuszczyk — APPROVED** [review]

**Pull request merged**